.github/gh-pages-publish.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Deploy Docs
+on:
+  push:
+    branches:
+      - master
+permissions:
+  contents: write
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2.2.4
+
+      - name: Install deps
+        run: pnpm install
+
+      - name: Install and Build 🔧
+        run: |
+          cd ./packages/mermaid
+          pnpm docs:build:vitepress
+
+      - name: Deploy 🚀📕
+        uses: JamesIves/github-pages-deploy-action@v4.3.3
+        with:
+          branch: gh-pages
+          folder: packages/mermaid/src/vitepress/.vitepress/dist
```

Comment:
deleted

Reply:
I also fixed the ts file so defaults values now also appears in the docs (In case this was the only problem)
It should be before a method to be used.

Reply:
> fixed the topis above 

Great! Thanks for the fixes!

> deleted the gh deploy yml file

I'm okay for having a `gh-deploy.yml` file in this PR if it saves maintainer's time. However, for a future PR, we should change the `JamesIves/github-pages-deploy-action` step to the official GitHub `actions/deploy-pages`.

> I also fixed the ts file so defaults values now also appears in the docs (In case this was the only problem)
> It should be before a method to be used.

It's better, but it's still missing quite a bit compared to the old https://mermaid-js.github.io/mermaid/#/Setup page.

E.g. these docs about the `logLevel` configuration option is nowhere on the new site: 
![image](https://user-images.githubusercontent.com/19716675/198914050-6a52e145-7e00-4e5a-bfd5-08f25a814a9b.png)

I think the reason why these docs are not getting exported is because the TSDoc comments are on a value, not on a type.

We might need to rewrite the TSDoc comments (or convert `config.ts` and `defaultConfig.ts` to use TypeScript classes instead of being a plain object) to get this exported properly to markdown. But this seems like a lot of work, so we should leave it to a future PR.

Unless somebody can think of a quick fix, for this PR, I think we should be okay with just adding a link to <https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/defaultConfig.ts>, and tell people that the full docs are there.